### PR TITLE
feature(test-cases): Add Debian 12 Artifact tests

### LINKED ISSUE #7852: Add support for Debian:12 in artifacts tests
Following the debian:10 EOL, we should add support for Debian:12 instead

### PR DESCRIPTION
**Pull request by k0machi**
This commit adds configuration and pipeline files for artifacts tests on
Debian 12.

Fixes #7852

### Testing

- [x] [Jenkins](https://jenkins.scylladb.com/job/scylla-staging/job/alexey/job/artifacts-debian12-test/)

### PR pre-checks (self review)

- [x] I added the relevant `backport` labels
- [x] I didn't leave commented-out/debugging code

### Reminders

- Add New configuration option and document them (in `sdcm/sct_config.py`)
- Add unit tests to cover my changes (under `unit-test/` folder)
- Update the Readme/doc folder relevant to this change (if needed)

### CODE REVIEW

**Comment by soyacz** [comment]
test failed, also ensure non-root test run

**Comment by k0machi** [comment]
> > test failed, also ensure non-root test run
> 
> Done, normal test succeeds without installing java, for offline I've added adoptium repo with temurin builds of java 11. Now it runs.
> 
> (For reference, `#6` is a normal run, `#13` and `#14` are offline and offline non-root runs)
Offline install is timing out, checking.

**Comment by fruch** [comment]
> > > test failed, also ensure non-root test run
> > 
> > 
> > Done, normal test succeeds without installing java, for offline I've added adoptium repo with temurin builds of java 11. Now it runs.
> > (For reference, `#6` is a normal run, `#13` and `#14` are offline and offline non-root runs)
> 
> Offline install is timing out, checking.

maybe it's related to:
https://github.com/scylladb/scylla-cluster-tests/pull/8190

**Comment by fruch** [comment]
let's rebase and find out if housekeeping is an issue or not

**Comment by k0machi** [comment]
> let's rebase and find out if housekeeping is an issue or not

Seems to be passing now.

File: sdcm/cluster.py [hunk 1]
```diff
@@ -5122,7 +5131,10 @@ def node_setup(self, node, verbose=False, **kwargs):  # pylint: disable=unused-a
             self.log.debug('Skip loader setup for using a prepared AMI')
             return
 
-        elif node.distro.is_debian10 or node.distro.is_debian11:
+        elif node.distro.is_debian12:
+            node.install_package(package_name='openjdk-17-jre openjdk-17-jre-headless')
```
Comment (fruch) [inline sdcm/cluster.py:5135]: why do we need this installed ? now that temurin-11-jre is installed ?

File: defaults/manager_versions.yaml [hunk 2]
```diff
@@ -2,32 +2,32 @@ manager_repos_by_version:
   "master_latest":
     centos7: 'https://downloads.scylladb.com/manager/rpm/unstable/centos/master/latest/scylla-manager.repo'
     centos8: 'https://downloads.scylladb.com/manager/rpm/unstable/centos/master/latest/scylla-manager.repo'
-    debian10: 'https://downloads.scylladb.com/manager/deb/unstable/unified-deb/master/latest/scylla-manager.list'
     debian11: 'https://downloads.scylladb.com/manager/deb/unstable/unified-deb/master/latest/scylla-manager.list'
+    debian12: 'https://downloads.scylladb.com/manager/deb/unstable/unified-deb/master/latest/scylla-manager.list'
```
Comment (fruch) [inline defaults/manager_versions.yaml:6]: I hate the way this file is managed, we don't really need to specify specific distro release like that for years now.

Comment (fruch) [reply defaults/manager_versions.yaml:6]: I think https://github.com/scylladb/scylla-cluster-tests/pull/8210 would do the trick

Comment (k0machi) [reply defaults/manager_versions.yaml:6]: I just changed debian10 to 12 there, if there's a better way now I can remove this.

File: sdcm/cluster.py (hunk 1 above)
Comment (k0machi) [reply sdcm/cluster.py:5135]: This is normal install and it's something I haven't removed - I'll remove it now.

File: defaults/manager_versions.yaml (hunk 2 above)
Comment (k0machi) [reply defaults/manager_versions.yaml:6]: Okay, I can rebase once that PR is merged.

Comment (k0machi) [reply defaults/manager_versions.yaml]: Done

File: test-cases/artifacts/debian12.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+gce_n_local_ssd_disk_db: 1
+instance_provision: "spot"
+instance_provision_fallback_on_demand: true
+logs_transport: 'ssh'
+n_db_nodes: 1
+n_loaders: 0
+n_monitor_nodes: 0
+nemesis_class_name: 'NoOpMonkey'
+scylla_linux_distro: 'debian-bookworm'
+scylla_apt_keys:
```
Comment (fruch) [inline test-cases/artifacts/debian12.yaml:16]: Please remove this, we should define those only in one place

**Review by fruch — CHANGES_REQUESTED** [review]

File: sdcm/mgmt/cli.py [hunk 4]
```diff
@@ -923,7 +923,7 @@ def __init__(self, manager_node):
         LOGGER.info("Initiating Scylla-Manager, version: {}".format(self.sctool.version))
         list_supported_distros = [Distro.CENTOS7,
                                   Distro.ROCKY8, Distro.ROCKY9,
-                                  Distro.DEBIAN10, Distro.DEBIAN11,
+                                  Distro.DEBIAN11, Distro.DEBIAN12,
```
Comment (fruch) [inline sdcm/mgmt/cli.py:926]: Damn, I've missed this silly list in my PR.

We need to remove this check as well, we should support any debian or rhel based distro, no need for extra checks

Comment (k0machi) [reply sdcm/mgmt/cli.py]: Replaced with a simple `is_blank_like` check.

File: test-cases/artifacts/debian12.yaml (hunk 3 above)
Comment (k0machi) [reply test-cases/artifacts/debian12.yaml]: Removed `scylla_apt_keys`

**Review by fruch — APPROVED** [review]
LGTM

**Comment by fruch** [comment]
@yaronkaikov one more example of assum new distro release support,
you might want to trigger those once we'll merge this

**Review by yaronkaikov — APPROVED** [review]

**Pull request merged**